R/utils.R

```diff
@@ -108,3 +108,31 @@ random_string <- function(size = NULL) {
 
   substr(res, 1L, size)
 }
+
+dims_to_rowcol <- function(x, as_integer = FALSE) {
```

Comment:
Maybe think about the reverse: `row_col_to_dim()`?  This would allow us to do:

```r
#' @param row,col,dims Start col and rows for data.  When `dims` is specified,
#'   `row` and `col` are ignored.
foo <- function(wb, row = NULL, col = NULL, dims = row_col_to_dim(row, col)) {
  # pass row and col directly to dims, prioritize over the individual params
  dims <- validate_dims(dims)
  dims$row
  dims$col
  ...
}
```

Reply:
indeed! nice, I like it. will look at it

Reply:
The `validate_dims()` here is where I'd imagine we could implement some of those `cellranger`-like things.  Then we have a good helper function to consistently use for any instances in which we need to establish cell ranges and not have to worry about passing integers or `"A1:C3"` specs or anything.

Reply:
The character/integer part was solely for our output. In `update_cell()` I allow to use dims to create data on a worksheet. Iirc I use rows and cols to identify areas that are available and must be created or can simply be modified. Though, in an ideal world, I would want to ditch most of `update_cell()`s code and use what is available in `wide_to_long()` and a do a full join afterwards. Updating the existing cells and adding the newly created. Though that will probably be a task for a rainy day in the future. I really dislike the duplicated code there. It's among the earliest newly created code for `openxlsx2` and... not the best :)

Reply:
Initial draft is added. Calling it a day